tools/bazar/services/EntryManager.php

```diff
@@ -573,6 +573,42 @@ public function create($formId, $data, $semantic = false, $sourceUrl = null)
             $this->mailer->notifyAdmins($data, true);
         }
 
+		/////////////////////////////////////// - yg
+		// Signup account validation process
+		// If the created entry has a user field, utilisateur_wikini is defined, and we consider we are creating an account
+
+		if (isset($GLOBALS['utilisateur_wikini']))
+		{
+			$vWiki = $this->wiki;				
+			$userManager = $this->userManager;
+
+			if ($vWiki->GetConfigValue ("signup_mail_activation") === "1")
+			{	
+				// Inactivate the account first
+			
+				$userManager->inactivateUser ($GLOBALS['utilisateur_wikini'], "" /* empty key */, true /* ignore key and force inactivation */);
+
+				if ($userManager->sendActivationLink ($GLOBALS['utilisateur_wikini']))
+				{										
+					// We redirect to a page indicating that a mail was sent to the user in order to activate his account
+						
+					$userManager->logout ();										
+					$redirectUrl = ($vWiki->GetConfigValue("base_url")) . "ActivationLinkSent";
+					header('Location: ' . $redirectUrl);
+					$this->wiki->exit();
+				}
+				// Problems while sending mail
+				else
+				{		
+					throw new \RuntimeException("Error sending activation mail. Please contact the website administrator.");
```

Comment:
```suggestion
					flash(_t('USERFIELD_EMAIL_ACTIVATION_ERROR'),'error');
```
je pense que lever une erreur est un peu trop restrictif ici.
On peut toujours créer le compte, la fiche associé mais juste indiquer un message à destination de la prochaine fois que l'usage affiche une page.

clé `USERFIELD_EMAIL_ACTIVATION_ERROR` à définir dans le bon fichier de langue